### ISSUE TITLE
Villages spawning up in the air

### ISSUE THREAD

**Issue by chezpaul** [Multiverse/Multiverse-Core#1768]
Hey there,
I thought I'd ask here, as I've asked everywhere else and no one seems to know.
Would it be Multiverse's doing to create villages up in the air like [this](https://www.dropbox.com/s/8qfhyycofaedd2c/2016-10-25_10.43.48.png?dl=0)?
When I create a singleplayer world, everything looks fine but when I create a normal world in multiverse, I always get these villages. They aren't all like this but it's about 40% of them.
Here's my server log: http://pastebin.com/ey0URdJN

In this log I'm running b719 but it does the same with b722.
I've also tried b691 which seems to be the last 1.7.10 compatible version.
I always update to the latest but this started showing up all of a sudden.
Thanks

EDIT:Villages are also often 2 or 3 together like in the screenshot. And lots of them all over the world.
How can I make them more spare?

**Comment by chezpaul** [comment]
For what it's worth, I put a bounty on this.  ;-)
https://www.bountysource.com/issues/38748653-villages-spawning-up-in-the-air

**Comment by FernFerret** [comment]
Wow, that's interesting, I've never seen that. However, the issue is most likely not with Multiverse, but rather the world generator that you're using (if you're using a custom one) or the server version. Multiverse just uses the Multiworld API and does not actually allow setting structure frequency/height.

However, that doesn't take into account other parameters that might be messing with world generation. How did you create the world? (via a command? `/mv create`, via an import? `/mv import`)

Here are some code snippets for your reference: 

We just pass along a boolean that says "yes, do make structures", or "no, don't make structures":
https://github.com/Multiverse/Multiverse-Core/blob/a64a8d96058a32e6179aa40bf9945d564b13c9a4/src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java#L247

This is the only structure generation parameter I see in the API. This doesn't mean that there aren't more parameters in the implementation, but rather they haven't been exposed (yet):
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/World.html#canGenerateStructures()

TLDR: To me (at this time), it looks like something going on with your server/world rather than a bug or feature to be implemented.

Questions:
* What server (if you are running a server) are you running?
* Could you post the output from `/mvv -g`

**Comment by chezpaul** [comment]
I generate my world with multiverse only, using /mv create [worldname] normal and that's it.
I tried using Terrain Control but that didn't work at all with 1.7.10. (They stopped upgrading their 1.7.10 version which has a crazy bug in it)
I'm running on a [Thermos 58](https://github.com/CyberdyneCC/Thermos/releases) server (new KCauldron) running Forge 1.7.10.1614
My /mvv -g dump is here: [http://pastebin.com/Gj0qunNm](http://pastebin.com/Gj0qunNm)
Thanks

**Comment by FernFerret** [comment]
Wow, your world list is very colorful. Is it all worlds that have this issue or just ones that have BIOMESOP or BETWEENLANDS or any of those crazy biomes? I believe the standard **Types** are: `NORMAL`, `FLAT`, `VERSION_1_1`, `LARGE_BIOMES`, `AMPLIFIED` and `CUSTOMIZED` and default **Environments** are: `NORMAL`, `NETHER` and `THE_END`. So I'm a little bit surprised to see other values there.

I'll admit, I've never heard of Thermos and definitely haven't heard of any of these types or environments.

I'm totally willing to grab their code and have a peek but I definitely won't get this done today.

```yaml
[{"MVWorld@446895432":{"Gen":"null","Type":"NORMAL","Env":"NORMAL","Name":"Reminiscence"}}, 
{"MVWorld@1195032558":{"Gen":"null","Type":"NORMAL","Env":"NORMAL","Name":"Meteorite"}}, 
{"MVWorld@85961816":{"Gen":"null","Type":"BIOMESOP","Env":"BETWEENLANDS","Name":"DIM20"}}, 
{"MVWorld@1398256985":{"Gen":"null","Type":"BIOMESOP","Env":"ENDOFTIME","Name":"DIM-112"}}, 
{"MVWorld@1037650948":{"Gen":"null","Type":"BIOMESOP","Env":"UNDERDARK","Name":"DIM-100"}}, 
{"MVWorld@1331895264":{"Gen":"null","Type":"BIOMESOP","Env":"THE_END","Name":"DIM1"}}, 
{"MVWorld@744236086":
{"Gen":"null","Type":"NORMAL","Env":"NETHER","Name":"Reminiscence_nether"}}, 
{"MVWorld@937825080":{"Gen":"null","Type":"BIOMESOP","Env":"EM","Name":"DIM2"}}, 
{"MVWorld@1500878304":{"Gen":"null","Type":"BIOMESOP","Env":"SPECTRE","Name":"DIM3"}}, 
{"MVWorld@784026534":{"Gen":"null","Type":"BIOMESOP","Env":"NETHER","Name":"DIM-1"}}, 
{"MVWorld@2043734410":{"Gen":"null","Type":"BIOMESOP","Env":"TWILIGHTFOREST","Name":"DIM7"}},
{"MVWorld@1348474450":{"Gen":"null","Type":"FLAT","Env":"NORMAL","Name":"Empirepolis"}}, 
{"MVWorld@2124962395":{"Gen":"null","Type":"BIOMESOP","Env":"OUTER","Name":"DIM-42"}}, 
{"MVWorld@193399547":{"Gen":"null","Type":"NORMAL","Env":"NORMAL","Name":"SurvivalGames5"}}, 
{"MVWorld@280299566":{"Gen":"null","Type":"NORMAL","Env":"NORMAL","Name":"world"}}, 
{"MVWorld@355911766":{"Gen":"null","Type":"NORMAL","Env":"NORMAL","Name":"SonicCraft"}}, 
{"MVWorld@2140509782":{"Gen":"null","Type":"BIOMESOP","Env":"ICEAGE","Name":"DIM-25"}}, 
{"MVWorld@24620183":{"Gen":"null","Type":"BIOMESOP","Env":"MIRROR","Name":"DIM-39"}}, 
{"MVWorld@1035027552":{"Gen":"null","Type":"BIOMESOP","Env":"TROPICRAFT","Name":"DIM-127"}}, 
{"MVWorld@240271871":{"Gen":"null","Type":"BIOMESOP","Env":"TORMENT","Name":"DIM-38"}}, 
{"MVWorld@2037491033":{"Gen":"null","Type":"NORMAL","Env":"NORMAL","Name":"Stampys"}}, 
{"MVWorld@1532302800":
{"Gen":"null","Type":"NORMAL","Env":"THE_END","Name":"Reminiscence_End"}}]
```

**Comment by chezpaul** [comment]
haha!, well most of these worlds I haven't visited yet and were not created by Multiverse.
Betweenland I haven't been yet
End of time, Underdark, Spectre, Outer, basically any wierd one, I didn't go to yet as I just started living in survival.
I was creating "Meteorite" in order to get a Druid Circle (betweenlands) to spawn in the Wastelands. That's when I started seeing all the villages.

Funny thing, is now, I'm trying to import a single world that I create with only 4 or 5 mods into multiverse. So I /mv import world normal.
The world loads fine.
I can mvtp to it fine and fly around in it... Villages will be set as they should. So the new chunks that get loaded are all with normal villages.
But when I mvtp back to the original overworld in my server, then all the blocks gets changed to something else and I can't mvtp back into the new world I just imported.

It's a fine line in figuring out which mods can be present when creating a world that can be imported in multiverse.

But any how, this shows that an imported world will create new chunks with proper villages.
Only a multiverse created world will give me those villages up in the air.
